HACKER/HACKER/Resources/Extensions/UITextField+Extension.swift

```diff
@@ -0,0 +1,43 @@
+//
+//  UITextField+Extension.swift
+//  HACKER
+//
+//  Created by 김지수 on 2022/02/22.
+//
+
+import Foundation
+import UIKit
+
+// MARK: - UITextField
+extension UITextField {
+  func setClearButton(with image: UIImage, mode: UITextField.ViewMode) {
+    let clearButton = UIButton(type: .custom)
+    clearButton.setImage(image, for: .normal)
+    clearButton.frame = CGRect(x: 0, y: 0, width: 35, height: 34)
+    clearButton.contentMode = .scaleAspectFit
+    clearButton.addTarget(self, action: #selector(UITextField.clear(sender:)), for: .touchUpInside)
+    self.rightView = clearButton
+    self.rightViewMode = mode
+  }
+  @objc private func clear(sender: AnyObject) {
+    self.text = ""
+  }
+  /// 클리어 버튼 클릭 시 텍스트필드 내용 삭제
+  func setClearNickNameButton(with image: UIImage, mode: UITextField.ViewMode) {
+    let clearButton = UIButton(type: .custom)
+    clearButton.setImage(UIImage(named: "xWhite"), for: .normal)
+    clearButton.frame = CGRect(x: 0, y: 0, width: 35, height: 34)
+    clearButton.contentMode = .scaleAspectFit
+    clearButton.addTarget(self, action: #selector(UITextField.clearText(sender:)), for: .touchUpInside)
+    self.rightView = clearButton
+    self.rightViewMode = mode
+  }
+  @objc private func clearText(sender: AnyObject) {
+    self.text = ""
+    let nicknameVC = NicknameViewController()
+    DispatchQueue.main.async {
+      nicknameVC.countTextLabel.text = "0/6"
+      
+    }
+  }
```

Comment:
뭔가 이게 textfield extension 이라서 함수 안에 
`let nicknameVC = NicknameViewController()`

이렇게 직접 NicknameViewController을 넣는거 대신 vc를 파라미터 같은걸로 받고 여러곳에서 재사용 가능하게 하면 좋을거같기도 해! @objc에서 어떻게 사용해야 되는지 찾아봐야겠담...

혹시 이게 그거였나 0/6으로 안되서 우리 그때 상당히 킹받았던.. 그것..? 
요거 해결된건가..?!

Reply:
@dlwns33 노웁.. 노 해결상태로 기억력이슈 때문에 방치됐다 ㅋ..👩🏻‍🦲

Reply:
이건 멋있게 해결하는 법이 있어

```suggestion
  @objc private func clearText(sender: AnyObject) {
    self.text = ""
    }
  }
```

이걸 이렇게 지워버리고 싶은 거잖아

그러면 얘네가 기본적으로 제공하는 clearButton을 쓰면서 그 이미지만 바꿔주면 돼
```
textField.clearButtonMode = .always
if let clearButton = textField.value(forKeyPath: "_clearButton") as? UIButton {
  clearButton.setImage(UIImage(named:"xBlack"), for: .normal)
}
```
이런 식으로 기본 제공되는 clearButton을 만든 다음에
그 버튼의 프로퍼티 중에 _clearButton을 찾아서 이미지를 바꿔
주는 방법이야

커스텀으로 정의했던 clearButton 로직을 다 지우고
저기 위에 새로운 코드로 바꿔주면 이런 예쁜 모습을 볼 수 있어

![Mar-23-2022 14-13-48](https://user-images.githubusercontent.com/71601985/159628582-d5ef2863-3db9-47f1-917e-61d7c0861d48.gif)